tests/Tokenizer/Analyzer/AttributeAnalyzerTest.php

```diff
@@ -129,4 +130,286 @@ public static function provideIsAttributeCases(): iterable
 
         yield [true, '<?php #[Foo("(")] class Bar {}'];
     }
+
+    /**
+     * @requires PHP 8.0
+     *
+     * @dataProvider provideGetAttributeDeclarationsCases
+     *
+     * @param array{0: string, 1: list<AttributeAnalysis>} $expected
+     */
+    public function testGetAttributeDeclarations(string $code, int $startIndex, array $expected): void
+    {
+        $tokens = Tokens::fromCode($code);
+        $attributeDeclarationAnalyses = AttributeAnalyzer::collectFor($tokens, $startIndex);
+
+        self::assertSame(
+            serialize($expected),
+            serialize($attributeDeclarationAnalyses),
+        );
+    }
+
+    /**
+     * @return iterable<array{0: string, 1: int, 2: list<AttributeAnalysis>}>
+     */
+    public static function provideGetAttributeDeclarationsCases(): iterable
+    {
+        yield [
```

Comment:
Please add this test case:

```suggestion
        yield [
            '<?php
            function foo() {}
            function bar() {}
            ',
            11,
            [],
        ];

        yield [
```

In general, I wanted `AttributeAnalyzer::collectFor()` to work for  all `Attribute::TARGET_*`. Ideally it should auto-rewind to the token that is first in the signature (e.g. when you call it for `class` it should skip `final`, `readonly` or `abstract`) and just look for the attributes defined for it. As a first iteration we can assume it's rewinded before the call, so we expect attributes _or_ any other meaningful token before the specified index.

That's why I would expect test cases here to use index of a token _after_ attributes, not the first `#[`.

Test case provided above should return empty collection because `bar()` does not have attributes defined. Now it rewinds to the beginning of the file and crashes 😅.

Reply:
Test added.